api/v1/checks.go

```diff
@@ -481,10 +483,113 @@ type AlertManager struct {
 	AlertManagerCheck `yaml:",inline" json:",inline"`
 }
 
+// CheckRelationshipLookup specifies the type of lookup to perform.
+type CheckRelationshipLookup struct {
```

Comment:
Can we make this + the Name, Namespace, Type tuple a generic lookup struct in duty, and then we can have
```go
type TemplateEnvironment interface {
GetEnvironment() map[string]any
}

duty.LookupComponent(ctx,  lookup)
duty.LookupConfig(ctx,  lookup)
```

We can then eventually replace all the existing lookup type use cases with the common Lookup struct and then have the option to add more type of matchers in the future that would apply everywhere?

Reply:
https://github.com/flanksource/duty/pull/359